corehq/util/context_processors.py

```diff
@@ -170,7 +170,7 @@ def get_demo(request):
     num_trial_days_remaining = 0
     if (settings.IS_SAAS_ENVIRONMENT
             and settings.ANALYTICS_IDS.get('HUBSPOT_API_ID')):
-        if not request.user.is_authenticated:
+        if getattr(request, 'user', None) and not request.user.is_authenticated:
```

Comment:
I think you could have used `hasattr` here since you are just checking it has the attribute

Reply:
lol, just saw @nickpell 's comment saying the opposite.